packages/roosterjs-editor-plugins/lib/plugins/Paste/sourceValidations/getPasteSource.ts

```diff
@@ -0,0 +1,47 @@
+import documentContainWacElements from './documentContainWacElements';
+import isExcelDesktopDocument from './isExcelDesktopDocument';
+import isGoogleSheetDocument from './isGoogleSheetDocument';
+import isPowerPointDesktopDocument from './isPowerPointDesktopDocument';
+import isWordDesktopDocument from './isWordDesktopDocument';
+import shouldConvertToSingleImage from './shouldConvertToSingleImage';
+import { BeforePasteEvent, ClipboardData, IEditor } from 'roosterjs-editor-types';
+import { KnownSourceType } from './KnownSourceType';
+
+/**
+ * @internal
+ */
+export type getSourceFunction = (
+    htmlAttributes: Record<string, string>,
+    fragment?: DocumentFragment,
+    editor?: IEditor,
+    clipboardData?: ClipboardData
```

Comment:
In what case these parameters are not passed? 

Suggest make all of them required. As an internal function, we'd better make it more clearly defined, rather than easier to call.